src/shell/shell_cmds.cpp

```diff
@@ -1390,7 +1455,15 @@ void DOS_Shell::CMD_DATE(char * args) {
 	}
 	// check if a date was passed in command line
 	uint32_t newday, newmonth, newyear;
-	if (sscanf(args, "%u-%u-%u", &newmonth, &newday, &newyear) == 3) {
+	int n = dos.tables.country[0] == 1
+	                ? sscanf(args, "%u%c%u%c%u", &newday, &c1, &newmonth,
+	                         &c2, &newyear)
+	                : (dos.tables.country[0] == 2
+	                           ? sscanf(args, "%u%c%u%c%u", &newyear, &c1,
+	                                    &newmonth, &c2, &newday)
+	                           : sscanf(args, "%u%c%u%c%u", &newmonth, &c1,
+	                                    &newday, &c2, &newyear));
+	if (n == 5 && c1 == c && c2 == c) {
```

Comment:
Even with clang formatting helping visually lay-out the ternary if statements - there's still far too much going on for my mind to fit it all in and see what's actually trying to be accomplished.

It's OK if this needs more lines to make it more self-documenting.

Reply:
I hope the updated code (with variables like date_format) is easier to read than the previous one.